File: BaeYoungSuk/Q25.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        Map<String, Integer> map = new HashMap<>(); 
        for(String[] clothArr : clothes){
            String type = clothArr[1];
            map.put(type, map.getOrDefault(type, 0) + 1);
        }
        
        for(Integer cloth : map.values()){
            answer *= cloth.intValue() + 1;
        }
        
        return answer - 1;
    }
}

File: kimssumin/Q25.py
def solution(clothes):
    answer = 1
    cl_dic = {}
    for c in clothes:
        if (c[1] in cl_dic):
            cl_dic[c[1]] += 1
        else:
            cl_dic[c[1]] = 1
    for case in cl_dic:
        answer *= (cl_dic[case]) + 1
    
    return answer - 1
